### PR TITLE
Tron transaction signing

### PR DESCRIPTION
**Pull request by johnnynanjiang**
Test case is in TronTransactionTests

### CODE REVIEW

File: Sources/Tron/Tron.pb.swift [hunk 1]
```diff
@@ -0,0 +1,4377 @@
+// DO NOT EDIT.
```
Comment (alejandro-isaza) [inline Sources/Tron/Tron.pb.swift]: Can we include the original protobuf file and a script to generate this instead?

File: Sources/Data+String.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Copyright © 2017-2018 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import Foundation
+
+extension Data {
+    public func toString() -> String {
```
Comment (alejandro-isaza) [inline Sources/Data+String.swift]: I don't think these extensions are useful. They don't provide any extra clarity on use and introduce an unexpected side effect of retiring empty string on error.

File: Sources/Data+String.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        return String(data: self, encoding: .utf8) ?? ""
+    }
+}
+
+extension String {
```
Comment (alejandro-isaza) [inline Sources/Data+String.swift]: Same here.

File: Sources/Tron/TronTransaction.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+import Foundation
+
+extension TronTransaction {
+    public var hasSignature: Bool {
+        return signature.count == 1
+    }
+
+    public mutating func addSignature(_ signature: Data) {
+        self.signature.append(signature)
```
Comment (alejandro-isaza) [inline Sources/Tron/TronTransaction.swift]: `self` is redundant here.

File: Sources/Tron/TronTransaction.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    public mutating func sign(privateKey: Data) -> Data {
+        let signature = Crypto.sign(hash: self.hash(), privateKey: privateKey)
+        self.addSignature(signature)
+        return signature
+    }
+}
+
+extension TronTransaction.RawData {
+    public mutating func toData() -> Data {
```
Comment (alejandro-isaza) [inline Sources/Tron/TronTransaction.swift]: I don't think this should be mutating, we should return a copy of the data.

**Comment by alejandro-isaza** [comment]
Please exclude `Tron.pb.swift` from linting in `.swiftlint.yml`

File: Sources/Data+String.swift (hunk 2 above)
Comment (johnnynanjiang) [reply Sources/Data+String.swift]: Yes they are redundant, will remove them, cheers.

File: Sources/Tron/TronTransaction.swift (hunk 5 above)
Comment (johnnynanjiang) [reply Sources/Tron/TronTransaction.swift]: That makes sense, will fix it.

File: Sources/Tron/Tron.pb.swift (hunk 1 above)
Comment (johnnynanjiang) [reply Sources/Tron/Tron.pb.swift]: I've updated README.md to include the steps

**Pull request closed without merging**